From cd2b658e83d47e8f49d1bbaf54be6433a21d3758 Mon Sep 17 00:00:00 2001
From: Luis Menina <liberforce@freeside.fr>
Date: Mon, 04 Mar 2013 17:10:26 +0000
Subject: Remove deprecated GnomeEntry.

It is replaced by GtkEntry. This means we lose the history
functionality, but that could be reimplemented using GtkComboBoxText
with an entry.

https://bugzilla.gnome.org/show_bug.cgi?id=589045
---
diff --git a/data/glade/sql.glade b/data/glade/sql.glade
index 2b12b85..f673ae0 100644
--- a/data/glade/sql.glade
+++ b/data/glade/sql.glade
@@ -165,24 +165,16 @@
 	  </child>
 
 	  <child>
-	    <widget class="GnomeEntry" id="server_entry">
+	    <widget class="GtkEntry" id="server_entry">
 	      <property name="visible">True</property>
-	      <property name="history_id">database_server</property>
-	      <property name="max_saved">10</property>
-
-	      <child internal-child="entry">
-		<widget class="GtkEntry" id="combo-entry1">
-		  <property name="visible">True</property>
-		  <property name="can_focus">True</property>
-		  <property name="editable">True</property>
-		  <property name="visibility">True</property>
-		  <property name="max_length">0</property>
-		  <property name="text" translatable="yes"></property>
-		  <property name="has_frame">True</property>
-		  <property name="invisible_char" translatable="yes">*</property>
-		  <property name="activates_default">True</property>
-		</widget>
-	      </child>
+	      <property name="can_focus">True</property>
+	      <property name="editable">True</property>
+	      <property name="visibility">True</property>
+	      <property name="max_length">0</property>
+	      <property name="text" translatable="yes"></property>
+	      <property name="has_frame">True</property>
+	      <property name="invisible_char" translatable="yes">*</property>
+	      <property name="activates_default">True</property>
 	    </widget>
 	    <packing>
 	      <property name="left_attach">1</property>
@@ -194,24 +186,16 @@
 	  </child>
 
 	  <child>
-	    <widget class="GnomeEntry" id="user_entry">
+	    <widget class="GtkEntry" id="user_entry">
 	      <property name="visible">True</property>
-	      <property name="history_id">database_user</property>
-	      <property name="max_saved">10</property>
-
-	      <child internal-child="entry">
-		<widget class="GtkEntry" id="combo-entry2">
-		  <property name="visible">True</property>
-		  <property name="can_focus">True</property>
-		  <property name="editable">True</property>
-		  <property name="visibility">True</property>
-		  <property name="max_length">0</property>
-		  <property name="text" translatable="yes"></property>
-		  <property name="has_frame">True</property>
-		  <property name="invisible_char" translatable="yes">*</property>
-		  <property name="activates_default">True</property>
-		</widget>
-	      </child>
+	      <property name="can_focus">True</property>
+	      <property name="editable">True</property>
+	      <property name="visibility">True</property>
+	      <property name="max_length">0</property>
+	      <property name="text" translatable="yes"></property>
+	      <property name="has_frame">True</property>
+	      <property name="invisible_char" translatable="yes">*</property>
+	      <property name="activates_default">True</property>
 	    </widget>
 	    <packing>
 	      <property name="left_attach">1</property>
@@ -248,24 +232,16 @@
 	  </child>
 
 	  <child>
-	    <widget class="GnomeEntry" id="db_entry">
+	    <widget class="GtkEntry" id="db_entry">
 	      <property name="visible">True</property>
-	      <property name="history_id">database_name</property>
-	      <property name="max_saved">10</property>
-
-	      <child internal-child="entry">
-		<widget class="GtkEntry" id="combo-entry3">
-		  <property name="visible">True</property>
-		  <property name="can_focus">True</property>
-		  <property name="editable">True</property>
-		  <property name="visibility">True</property>
-		  <property name="max_length">0</property>
-		  <property name="text" translatable="yes"></property>
-		  <property name="has_frame">True</property>
-		  <property name="invisible_char" translatable="yes">*</property>
-		  <property name="activates_default">True</property>
-		</widget>
-	      </child>
+	      <property name="can_focus">True</property>
+	      <property name="editable">True</property>
+	      <property name="visibility">True</property>
+	      <property name="max_length">0</property>
+	      <property name="text" translatable="yes"></property>
+	      <property name="has_frame">True</property>
+	      <property name="invisible_char" translatable="yes">*</property>
+	      <property name="activates_default">True</property>
 	    </widget>
 	    <packing>
 	      <property name="left_attach">1</property>
diff --git a/src/planner-sql-plugin.c b/src/planner-sql-plugin.c
index 14a088c..ea2f46a 100644
--- a/src/planner-sql-plugin.c
+++ b/src/planner-sql-plugin.c
@@ -27,7 +27,6 @@
 #include <stdlib.h>
 #include <glib.h>
 #include <glib/gi18n.h>
-#include <libgnomeui/gnome-entry.h>
 #include <glade/glade.h>
 #include <gtk/gtk.h>
 #include <libgda/libgda.h>
@@ -950,12 +949,9 @@ sql_plugin_retrieve_db_values (PlannerPlugin  *plugin,
 
 	gtk_window_set_title (GTK_WINDOW (dialog), title);
 
-	server_entry = gnome_entry_gtk_entry (
-		GNOME_ENTRY (glade_xml_get_widget (gui, "server_entry")));
-	db_entry = gnome_entry_gtk_entry (
-		GNOME_ENTRY (glade_xml_get_widget (gui, "db_entry")));
-	user_entry = gnome_entry_gtk_entry (
-		GNOME_ENTRY (glade_xml_get_widget (gui, "user_entry")));
+	server_entry   = glade_xml_get_widget (gui, "server_entry");
+	db_entry       = glade_xml_get_widget (gui, "db_entry");
+	user_entry     = glade_xml_get_widget (gui, "user_entry");
 	password_entry = glade_xml_get_widget (gui, "password_entry");
 
 	str = planner_conf_get_string (CONF_SERVER, NULL);
--
cgit v0.9.2
